src/game/GamePlay.hxx

```diff
@@ -2,13 +2,15 @@
 #define GAMEPLAY_HXX_
 
 #include "ZoneManager.hxx"
+#include "Singleton.hxx"
 
-class GamePlay
+class GamePlay : public Singleton<GamePlay>
 {
 public:
   GamePlay() = default;
 
   void update();
+  ZoneManager *getZoneManager() { return &m_ZoneManager; };
```

Comment:
```suggestion
  ZoneManager &getZoneManager() { return m_ZoneManager; };
```
I would prefer to return the reference here.

Reply:
ok, done